randovania/games/planets_zebeth/exporter/patch_data_factory.py

```diff
@@ -43,14 +43,24 @@ def _create_pickups_dict(self, pickup_list: list[ExportedPickupDetails], item_in
                 else 0
             )
 
+            pickup_type = "Nothing"
+            if not pickup.other_player:
+                if pickup.original_pickup.name.startswith("Tourian Key"):
+                    pickup_type = "Tourian Key"
+                else:
+                    pickup_type = pickup.original_pickup.name
+
+            acquired_msg = pickup.name
+            if not self.players_config.is_multiworld:
+                acquired_msg = f"{pickup.original_pickup.name} acquired"
+
             pickup_map_dict[object_id] = {
+                "index": pickup.index.index,
```

Comment:
don't rely on the pickup index. if you need it, add the index as extra data

Reply:
the pickup index is either not referenced and null on patcher side then the patcher gets the pickup index from object id
or the pickup index is referenced and used

Reply:
When i said "dont rely on the pickup index" i didn't meant patcher side, I meant rdv side. 
If you need the pickup index on the patcher side, thats fine, but on the rdv side use extra data for it, rather than the index of the pickup node.

Reply:
so do I drop the "index" key value pair?

Reply:
no, you just do something like this instead:
`"index": self.game.region_list.node_from_pickup_index(pickup.index).extra["item_index_how_its_used_in_planets"]`

Reply:
unless you want me to readd "pickup_index" as extra on each pickups. as I said `"index"` is optional since the patcher can retrieve the index from object id